modifierGui.py

```diff
@@ -0,0 +1,154 @@
+import re
+import json
+from alpacaModifier import AlpacaModifier
+
+# The regex
+REGEX_INSTRUCTION = r'^.*$'
+REGEX_INPUT = r'^([\d\^\+\-\/yx=,\(\) ]|\\u00b\d)+$'
+REGEX_OUTPUT = r'^.*$'
+
+SAVE_FILE = 'alpaca_new.json'
+
+class MyAlpacaModifier(AlpacaModifier):
+	def fill_values(self, instruction='', input='', old_output='', modified_output=''):
+		self.instruction = instruction
+		self.input = input
+		self.old_output = old_output
+		self.modified_output = modified_output
+
+	def save_prev_item(self):
+		if self.prev_item != None:
+			self.prev_item['instruction'] = self.instruction
+			self.prev_item['input'] = self.input
+			self.prev_item['output'] = self.modified_output
+			print(f'saving as:\n{self.prev_item}')
+			self.prev_item = None
+
+	def next_generator(self):
+		for item in self.data:
+			self.save_prev_item()
+			
+			match_instruction = re.search(self.pattern_instruction, item['instruction'])
+			match_input = re.search(self.pattern_input, item['input'])
+			match_output = re.search(self.pattern_output, item['output'])
+			if match_instruction and match_input and match_output:
```

Comment:
This returns false on empty "input"

Reply:
Indeed. In the above lines, we match the item based on `REGEX_INSTRUCTION`, `REGEX_INPUT`, and `REGEX_OUTPUT`. I've defined `REGEX_INPUT = r'^([\d\^\+\-\/yx=,\(\) ]|\\u00b\d)+$'` so that it matches inputs that have only digits and math operators/variables.
To enable matching of empty string, you can instead do `REGEX_INPUT = r'^.*$'` (matches everything).
This file is just an example of how to utilize `AlpacaModifier`. Users can use it and modify it to match their needs, or write a new one.